crypto/bn/bn_exp.c

```diff
@@ -615,6 +615,15 @@ int BN_mod_exp_mont_consttime(BIGNUM *rr, const BIGNUM *a, const BIGNUM *p,
 
     top = m->top;
 
+    if (in_mont != NULL && BN_is_zero(&in_mont->N)) {
+        ERR_raise(ERR_LIB_BN, ERR_R_PASSED_INVALID_ARGUMENT);
+        return 0;
+    }
+    if ((unsigned int)top > INT_MAX / sizeof(m->d[0]) / (1 << 8)) {
```

Comment:
Is 1<<8 some arbitrary number?

If I read things correctly, the maximum size we'll allocate is 9*top*sizeof(m->d[0]).

Reply:
window can be up to 6 and numpowers is then 1<<6. And we add 2*top to that and some constant. Yeah 1<<7 would be probably sufficient but I do not think it really matters whether we allow 16MB bignums or just 8MB ones.

Reply:
8 MB bignums are already way bigger than one anybody should be using.